File: binderBack/src/main/java/com/binder/entity/Book.java
package com.binder.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name="Book")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Book {
    @Id
    @GeneratedValue
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name="author", nullable = false)
    private String author;

    @Column(name = "year")
    private Integer year;

    @Column(name = "description", nullable = false)
    private String description;
}

File: Binder/app/src/main/java/com/example/binder/model/Book.kt
package com.example.binder.model

import com.squareup.moshi.Json

data class Book(
    @field:Json(name = "id") val id: Long?,
    @field:Json(name = "title") val title: String,
    @field:Json(name = "author") val author: String,
    @field:Json(name = "year") val year: Int?,
    @field:Json(name = "description") val description: String
)

fun Book.toInfoStr(): String = if (year != null) "$author/$title ($year)" else "$author/$title"